src/OmniSharp.Roslyn/BufferManager.cs

```diff
@@ -128,32 +128,35 @@ private bool TryAddTransientDocument(string fileName, string fileContent)
             var projects = FindProjectsByFileName(fileName);
             if (!projects.Any())
             {
-                return false;
+                //todo: deal with the deletion of the misc files
+                _workspace.TryAddMiscellaneousDocument(fileName, LanguageNames.CSharp);
```

Comment:
Take a look at the OnWorkspaceChanged handler in this class--you'll want to augment it to remove misc files the way we remove transient documents when they are deleted.

Reply:
(This will fix the bug you mentioned)